invenio_rdm_records/resources/serializers/csl/__init__.py

```diff
@@ -18,3 +36,89 @@ class CSLJSONSerializer(MarshmallowJSONSerializer):
     def __init__(self, **options):
         """Constructor."""
         super().__init__(schema_cls=CSLJSONSchema, **options)
+
+
+class StringCitationSerializer(CSLJSONSerializer):
+    """CSL Citation Formatter serializer for records.
+
+    In order to produce a formatted citation of a record through citeproc-py,
+    we need a CSL-JSON serialized version of it.
+    """
+
+    _default_style = 'harvard1'
+    """The `citeproc-py` library supports by default the 'harvard1' style."""
+
+    _default_locale = 'en-US'
+    """The `citeproc-py` library supports by default the 'harvard1' style."""
+
+    _user_args = {
+        'style': fields.Str(missing=_default_style),
+        'locale': fields.Str(missing=_default_locale)
+    }
+    """Arguments for the webargs parser."""
+
+    _valid_formats = ('csl', 'bibtex')
+    """Supported formats by citeproc-py."""
+
+    def __init__(self, record_format='csl'):
```

Comment:
This makes is fully schema dependent. When we talked about it, it seemed like a good option... Now I am wondering if we should inherit from `Marshmallow....` and receive the `schema` parameter (defaulting to `CSLJSONSchema`). WDYT?

Reply:
It depends, is there any option that we want the String Citation Serializer to be able to use different schemas?

Reply:
If `CSLJSONSerializer` is not needed as parent class, I would remove it. The code is using `self.dump_one`, should it use it from `CSLJSONSerializer` or from `Marshmallow...`?